HeterogeneousCore/CUDAUtilities/interface/launch.h

```diff
@@ -51,11 +51,11 @@ namespace cms {
       size_t sharedMem;
       cudaStream_t stream;
 
-      LaunchParameters(dim3 gridDim, dim3 blockDim, size_t sharedMem = 0, cudaStream_t stream = nullptr)
-          : gridDim(gridDim), blockDim(blockDim), sharedMem(sharedMem), stream(stream) {}
+      LaunchParameters(dim3 lgridDim, dim3 lblockDim, size_t lsharedMem = 0, cudaStream_t lstream = nullptr)
+          : gridDim(lgridDim), blockDim(lblockDim), sharedMem(lsharedMem), stream(lstream) {}
 
-      LaunchParameters(int gridDim, int blockDim, size_t sharedMem = 0, cudaStream_t stream = nullptr)
-          : gridDim(gridDim), blockDim(blockDim), sharedMem(sharedMem), stream(stream) {}
+      LaunchParameters(int lgridDim, int lblockDim, size_t lsharedMem = 0, cudaStream_t lstream = nullptr)
+          : gridDim(lgridDim), blockDim(lblockDim), sharedMem(lsharedMem), stream(lstream) {}
```

Comment:
Is there any way to tell the compiler "please stop bothering me about this, I know what I am doing and I do like to keep my variable names like they are, thank you very much ?" - other than disabling the warning altogether ?

It's not like a variable can be constructed from itself.

Reply:
@fwyzard , I am not an expert but why you need gridDim(gridDim) ? From pedestrian point of view, the name of parameters in a class constructor does not matter, what is matter for a code are names of class members. May be worse to check what gcc experts are thinking and why they introduced this warning. They use to add new warnings, which are not obvious to users, but usually have some arguments (may be very formal).

Reply:
The name of any single variable does not matter, the compiled code will happily work in the same way if we call them all `aaa`, `aab`, `aac`, _etc_.

Of course the code will be very hard to read and follow in that case...

So, yes, we could name the parameters
```c++
aunchParameters(dim3 par1, dim3 par2, size_t par3 = 0, cudaStream_t par4 = nullptr)
          : gridDim(par1), blockDim(par2), sharedMem(par3), stream(par4) {}
```
Then the code would compile, and `-Wshadow` would not trigger any exception.

Of course the code would be harder to read and maintain - so we give parameters names that make it easier to read and maintain the code.

If you call the parameter `lgridDim` - what does the `l` stand for ?

To improve the naming in that class one could
  - rename all member variables from `gridDim` to `gridDim_`;
  - add accessors for `gridDim_` named `gridDim()`;
  - rewrite the constructor to take a parameter named `gridDim` and use it to initialise the data member `gridDim_`.

Reply:
well `LaunchParameters() : gridDim(gridDim) {}` will still compile (with a warning about uninitialized variable)

Reply:
> If you call the parameter `lgridDim` - what does the `l` stand for ?
> 
> To improve the naming in that class one could
> 
>     * rename all member variables from `gridDim` to `gridDim_`;
> 
>     * add accessors for `gridDim_` named `gridDim()`;
> 
>     * rewrite the constructor to take a parameter named `gridDim` and use it to initialise the data member `gridDim_`.

sure, `l` prefix is somewhat random (I have it to mean local).

@fwyzard 
Would `gridDim_l` (or `_a` for arg) keep the code still readable in your opinion?

The option to rewrite the class to change members to follow CMSSW naming rules is also possible, but would require more effort.